snapcraft/internal/project_loader.py

```diff
@@ -299,10 +303,34 @@ def _build_env(root, arch_triplet):
         for envvar in ['CPPFLAGS', 'CFLAGS', 'CXXFLAGS']:
             env.append(formatting_utils.format_path_variable(
                 envvar, paths, prepend='-I', separator=' '))
+
+    if confinement == 'classic':
+        if not core_linker:
+            raise EnvironmentError('classic confinement requires the '
+                                   'core_linker to be set')
+
+        core_path = os.path.join('/snap', 'core', 'current')
```

Comment:
Note that this is still `ubuntu-core` in the most recent snapd available in xenial (2.16ubuntu3). We probably want to support both unless we want a hard dependency on a newer snapd version.

In a similar vein, have we considered how we might we support other cores? I assume they'd be named differently (thus mounted in a different location).